tools/bridge/bridge/transfer_recorder.py

```diff
@@ -33,6 +39,28 @@ def __init__(self, minimum_balance: int) -> None:
 
         self.is_validator: Optional[bool] = None
         self.balance: Optional[int] = None
+        self.last_fetcher_reached_head_event: Dict[
+            ChainRole, FetcherReachedHeadEvent
+        ] = {}
+
+    def get_state_summary(self):
```

Comment:
Conceptually, I think this would fit better in the `InternalState`. Then the job of the `InternalState` would be to present what the application is doing and the job of the `TransferRecorder` is actually do stuff. Right now the `TransferRecorder` has to do both. Also, in the future we could add other data sources that the `InternalState` can read from and then we'd have presentation logic spread over the whole application.

Reply:
I kind of see your point. But, on the other hand we would have to reach into the transfer_recorder from the internal state object.

> Also, in the future we could add other data sources that the InternalState can read from

We will do this.

My suggestion then would be to add a 

```
@functools.singledispatch
def get_internal_state_summary(obj):
```
and implement that for each object which we want to publish via the `InternalState` class. 
That would mean we still spread `get_internal_state_summary` over the whole application, but in a dedicated function and we can implement that for each class that we want to without changing the class. And the `InternalState` only has to call that function and not reach into the other classes.
 
 
Would that be fine for you?